<a href="https://colab.research.google.com/github/HansErikLima/mandatory_assignment-2_machine_learning/blob/main/Machine_Learning_Assignment_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# IDATG2208 - Mandatory Assignment - 2


In [6]:
from sklearn.datasets import load_breast_cancer
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

## Exercise 1: Data Preparation

## Q1.1
Load the dataset, inspect feature names and target distribution. Comment on
dataset imbalance.

In [12]:
# Q1.1 - Load dataset and inspect features and target

# Load the Breast Cancer Wisconsin dataset
bc = load_breast_cancer()

# Convert to a DataFrame for easier inspection
X = pd.DataFrame(bc.data, columns=bc.feature_names)
y = pd.Series(bc.target, name="target")

# Show number of samples and features
print("Shape of dataset:", X.shape)

# Show feature names
print("\nFeature names:")
print(bc.feature_names)

# Show target values and class names
print("\nTarget class names:", bc.target_names)
print("Unique target values:", np.unique(y))

Shape of dataset: (569, 30)

Feature names:
['mean radius' 'mean texture' 'mean perimeter' 'mean area'
 'mean smoothness' 'mean compactness' 'mean concavity'
 'mean concave points' 'mean symmetry' 'mean fractal dimension'
 'radius error' 'texture error' 'perimeter error' 'area error'
 'smoothness error' 'compactness error' 'concavity error'
 'concave points error' 'symmetry error' 'fractal dimension error'
 'worst radius' 'worst texture' 'worst perimeter' 'worst area'
 'worst smoothness' 'worst compactness' 'worst concavity'
 'worst concave points' 'worst symmetry' 'worst fractal dimension']

Target class names: ['malignant' 'benign']
Unique target values: [0 1]


In [10]:
# Q1.1 - Target distribution

class_counts = y.value_counts()
print("\nTarget distribution (counts):")
print(class_counts)

class_proportion = class_counts / len(y)
print("\nTarget distribution (proportion):")
print(class_proportion.round(4))


Target distribution (counts):
target
1    357
0    212
Name: count, dtype: int64

Target distribution (proportion):
target
1    0.6274
0    0.3726
Name: count, dtype: float64


### Q1.1 – Dataset Inspection and Target Distribution

The dataset contains 569 samples and 30 features.  
The features represent various measurements computed from breast tissue images.

By inspecting the target classes using `bc.target_names` and the unique values in `y`,  
we observe that the dataset uses two classes:
- 0 → malignant (cancerous)
- 1 → benign (non-cancerous)

The target distribution shows that benign cases are more frequent than malignant cases,  
indicating a moderate class imbalance. This is important to consider in later model evaluation.
